src/fairseq2/datasets/_data_reader.py

```diff
@@ -85,12 +85,14 @@ def __init__(
         pipeline: DataPipeline,
         gang: Gang,
         options: DataReadOptions,
+        strict_state: bool = True,
```

Comment:
@cbalioglu : i'm also adding this extra argument here, if it's ok for you ?

Reply:
yep, looks good. Just a nit preference; we can make it a keyword-only parameter considering it is optional.

```
def __init__(self, ..., *, strict_state: bool = True) -> None:
```